src/esrf_ontologies/technique/types.py

```diff
@@ -55,6 +58,27 @@ def _get_nxnote(self) -> Dict[str, Union[List[str], str]]:
             "iris": iris,
         }
 
+    def _fill_nxnote(self, nxnote: MutableMapping) -> None:
+        try:
+            names = nxnote["names"]
+        except KeyError:
+            names = list()
+        try:
+            iris = nxnote["iris"]
+        except KeyError:
+            iris = list()
+        techniques = dict(zip(iris, names))
+        for technique in self.techniques:
+            techniques[technique.iri] = technique.names[0]
```

Comment:
[minor] we could add a function `get_primary_name(technique)` that returns `technique.names[0]` to clearly indicate which name is selected.